flink-tests/src/test/java/org/apache/flink/test/state/operator/restore/keyed/KeyedJob.java

```diff
@@ -98,10 +98,9 @@ public static void main(String[] args) throws Exception {
 			.map(new StatefulStringStoringMap(mode, "first"))
 			.setParallelism(4);
 
-		// TODO: re-enable this when generating the actual 1.2 savepoint
-		//if (mode == ExecutionMode.MIGRATE || mode == ExecutionMode.RESTORE) {
-		map.uid("first");
-		//}
+		if (mode == ExecutionMode.MIGRATE || mode == ExecutionMode.RESTORE) {
```

Comment:
I somehow don't like it that is not explained in the commit message what has actually changed/why was this change required at all. Especially since you have not changed anything else in the code, it is difficult to understand that.  If nothing else has changed, why do we need this `if (...)`? If something has changed, shouldn't it be covered by some test?